### PR TITLE
2455 add protected areas section

### PR DESCRIPTION
**Pull request by yaguzmang**
Solves #2455 by:

Refactoring:
-  Dividing the logic to handle map layers: separating the layers handlers in different files, so they are more modular and easier to maintain.
- Renaming "layers" to "forestLayers" since there will be new non-forest layers (this is also done in the redux reducers).

Adding new logic and UI:
- Added a new hook, redux state management, and UI components for the protected areas.

TODO: Find a way to render the WDPA layer. After spending quite some time trying to get the WDPA layers to work, it seems like it is not possible the way it is set up now.

As per the [documentation](https://developers.google.com/earth-engine/guides/featureview_overview): "The process for visualizing a FeatureView asset using the Google Maps API is a four part process: obtain a tiles key using ee.data.getFeatureViewTilesKey, pass it to an instance of FeatureViewTileSource, construct an ee.layer.ImageOverlay [...]" at the moment we call `ee.data.getFeatureViewTilesKey` in the backend and then we send the response to the frontend. In the mapController we then try to pass it to a new `FeatureViewTileSource`, but due to the typing that is lost when sending the object from backend to frontend, the `FeatureViewTileSource` constructor doesn't recognise the object and throws an error. Unfortunately, there is no casting available for this kind of object.  

Large GIF (862x484)

### CODE REVIEW

**Review by eraviolo — DISMISSED** [review]

File: src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/ProtectedAreaLayerSection/CustomAssetControl/CustomAssetControl.scss [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    &:disabled {
+      background-color: #f8f7f7;
+      border-color: #dedfdf;
+      box-shadow: 0 0 10px 2px #efeeee inset;
+      cursor: default;
+    }
+  }
+
```
Comment (minotogna) [inline src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/ProtectedAreaLayerSection/CustomAssetControl/CustomAssetControl.scss]: Any chance to use the global color variables ?

File: src/client/store/ui/geo/slice.ts [hunk 2]
```diff
@@ -174,9 +182,57 @@ export const geoSlice = createSlice({
...
+      state.protectedAreasOptions.failedLayers = initialState.protectedAreasOptions.failedLayers
+    },
+    resetSingleProtectedAreaLayerStates: (state, { payload }: PayloadAction<ProtectedAreaKey>) => {
+      delete state.protectedAreasOptions.fetchedLayers[payload]
+      delete state.protectedAreasOptions.pendingLayers[payload]
+      delete state.protectedAreasOptions.failedLayers[payload]
+    },
+    setCustomProtectedAreaAssetId: (state, { payload }: PayloadAction<string>) => {
+      state.protectedAreasOptions.customAssetId = payload
+    },
```
Comment (minotogna) [inline src/client/store/ui/geo/slice.ts:235]: This file is becoming too big , that means there's either common logic to refactor or we should split this into several files ... what do you think ?

File: src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/ProtectedAreaLayerSection/CustomAssetControl/CustomAssetControl.scss (hunk 1 above)
Comment (yaguzmang) [reply src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/ProtectedAreaLayerSection/CustomAssetControl/CustomAssetControl.scss]: Yes, fixed this and a bug in the new commit.

File: src/client/store/ui/geo/slice.ts (hunk 2 above)
Comment (yaguzmang) [reply src/client/store/ui/geo/slice.ts:235]: Yes, I agree it is getting too big. The logic for the layers is similar but not always the same, so generalizing that logic would end up in a lot of extra code handling edge cases. I think it should better be split. Is it okay to do it in the upcoming Burned Areas PR?

Comment (minotogna) [reply src/client/store/ui/geo/slice.ts:235]: let's think how to refactor this later on?

Comment (yaguzmang) [reply src/client/store/ui/geo/slice.ts:235]: Yes, splitting it into files is a good start, but there should be a better way to handle the layers. We can take a look into that later.

Comment (minotogna) [reply src/client/store/ui/geo/slice.ts:235]: well, the best way would be to avoid duplicated code.

**Review by minotogna — APPROVED** [review]

**Review by eraviolo — APPROVED** [review]

**Pull request merged**